_posts/2018-10-25-Rust-1.30.md

```diff
@@ -260,25 +263,29 @@ mod baz {
     pub fn qux() {
         // old
         ::foo::bar();
+        // does not work, which is different than with `use`:
+        // foo::bar();
 
         // new
         crate::foo::bar();
     }
 }
 ```
 
-This will hopefully make absolute paths a bit more clear, and remove some of
-the ugliness of leading `::`.
+Once this style becomes widely used, this will hopefully make absolute paths a bit more
+clear and remove some of the ugliness of leading `::`.
 
-All of these changes combined lead to a more straightforward understanding of
-how paths resolve. When you see a path like `a::b::c`, you can ask:
+All of these changes combined lead to a more straightforward understanding of how paths
+resolve. Wherever you see a path like `a::b::c` someplace other than a `use` statement,
+you can ask:
 
 * Is `a` the name of a crate? Then we're looking for `b::c` inside of it.
 * Is `a` the keyword `crate`? Then we're looking for `b::c` from the root of our crate.
 * Otherwise, we're looking for `a::b::c` from the current spot in the module hierarchy.
 
-Since these rules apply uniformly everywhere, you'll need to tweak your
-imports much less when moving code around.
+The old behavior of `use` paths always starting from the crate root still applies. But
+after making a one-time switch to the new style, these rules will apply uniformly to
+paths everywhere, and you'll need to tweak your imports much less when moving code around.
```

Comment:
I don't know what this paragraph is talking about, the changes stabilized in 1.30 neither reduce "import tweaking when moving code around" in any way, nor "lead to a more straightforward understanding of how paths resolve".

All of those are supposed benefits of the 2018 edition module system, that's not stabilized yet and require explicit switching to 2018 edition.
Right now (2015 edition) `crate::` in paths is supported to simplify migration to 2018 rather than anything else.

Reply:
A bit more constructively:
- *Non*-imports starting with a crate name require less tweaking when moving code around due to the "`::serde_json::from_str` -> `serde_json::from_str`" change.
- If `crate::` is used consistently in imports (the "still on 2015, but ready for 2018 migration" mode), then we have more straightforward *import* logic - they always start with a crate name or a special keyword.

Reply:
i was purely talking about moving code within submodules, not moving to or from 2015/2018, like your first bullet point